Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs

```diff
@@ -253,6 +251,80 @@ private object OnBegin(object thisObj)
         }
 
         /// <summary>
+        ///  Common helper for all EndAsync Callbacks.
+        /// </summary>
+        /// <param name="taskObj">Returned task by the async method.</param>
+        /// <param name="thisObj">This object.</param>
+        private void OnEndAsync(object taskObj, object thisObj)
+        {
+            try
+            {
+                Task task = taskObj as Task;
+                if (task == null)
+                {
+                    DependencyCollectorEventSource.Log.NotExpectedCallback(0, "OnEndAsyncSql", "task == null");
+                    return;
+                }
+
+                DependencyCollectorEventSource.Log.EndAsyncCallbackCalled(thisObj.GetHashCode().ToString(CultureInfo.InvariantCulture));
+
+                task.ContinueWith(t =>
+                {
+                    Exception exceptionObj = null;
+                    if (t.IsFaulted)
+                    {
+                        exceptionObj = t.Exception != null && t.Exception.InnerException != null ? t.Exception.InnerException : t.Exception;
+                    }
+
+                    this.OnEnd(exceptionObj, thisObj);
+                });
+            }            
+            catch (Exception ex)
+            {
+                DependencyCollectorEventSource.Log.CallbackError(thisObj == null ? 0 : thisObj.GetHashCode(), "OnEndAsyncSql", ex);
+            }
+        }
+
+        /// <summary>
+        ///  Common helper for all EndAsync Callbacks that should send data only in the case of exception happened.
+        /// </summary>
+        /// <param name="taskObj">Returned task by the async method.</param>
+        /// <param name="thisObj">This object.</param>
+        private void OnEndExceptionAsync(object taskObj, object thisObj)
```

Comment:
there's OnEndAsync already, let's leave as is? comment is pretty self explanatory.

---
In reply to: [98070645](https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/298#discussion_r98070645) [](ancestors = 98070645)